From c59d5ddc80040da68aa30a4e18b6058abfce8416 Mon Sep 17 00:00:00 2001
From: Jiajie Chen <c@jia.je>
Date: Thu, 5 Aug 2021 22:36:25 +0800
Subject: [PATCH 07/11] binutils: fix assert op check
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

When building with gcc 10.2.1, the build fails with:

```
elfnn-loongarch.c: In function ‘loongarch_elf_relocate_section’:
elfnn-loongarch.c:1470:29: error: ‘opr1’ may be used uninitialized in this function [-Werror=maybe-uninitialized]
elfnn-loongarch.c:1445:11: note: ‘opr1’ was declared here
cc1: all warnings being treated as errors
```

This commit fixes incorrect logic to become: `loongarch_pop` does not succeed or the assertion check failed.
---
 bfd/elfnn-loongarch.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/bfd/elfnn-loongarch.c b/bfd/elfnn-loongarch.c
index 6d135da2342..873bba96208 100644
--- a/bfd/elfnn-loongarch.c
+++ b/bfd/elfnn-loongarch.c
@@ -1467,7 +1467,7 @@ perform_relocation (const Elf_Internal_Rela *rel, bfd_vma value,
 
     case R_LARCH_SOP_ASSERT:
       r = loongarch_pop (&opr1);
-      if (r != bfd_reloc_ok && opr1 == false)
+      if (r != bfd_reloc_ok || opr1 == false)
         r = bfd_reloc_notsupported;
       break;
 
-- 
2.30.1

